agent-inject/agent/container_sidecar.go

```diff
@@ -214,7 +214,7 @@ func (a *Agent) createLifecycle() corev1.Lifecycle {
 		flags := a.vaultCliFlags()
 		flags = append(flags, "-self")
 
-		lifecycle.PreStop = &corev1.Handler{
+		lifecycle.PreStop = &corev1.LifecycleHandler{
```

Comment:
This was apparently changed from Handler to LifecycleHandler in 0.23: https://github.com/kubernetes/api/commit/cca52a076791d31b1796453ece4b2f4df97406a2#diff-04e78d7bfddcb8b0b0fb2b950dbefcceb0af25bd827bfb41721f1280810a160aL2386